doc/developers_guide.rst

```diff
@@ -827,6 +827,44 @@ a valid Fortran program.
    # +++++++++++++
    # TBD
 
+.. _tokenisation:
+
+Tokenisation
+++++++++++++
+
+In order to simplify the problem of parsing code containing
+potentially complex expressions, fparser2 performs some limited
+tokenisation of a string before proceeding to attempt to match it.
+Currently, this tokenisation replaces three different types of quantity with
+simple names:
+
+ 1. the content of strings;
+ 2. expressions in parentheses;
+ 3. literal constants involving exponents (e.g. ``1.0d-3``)
+
+This tokenisation is performed by the `string_replace_map` function:
+
+.. autofunction:: fparser.common.splitline.string_replace_map
```

Comment:
I get this when building the documentation ... Could you please check.

/home/rupert/proj/PSyclone/external/fparser/src/fparser/common/splitline.py:docstring of fparser.common.splitline.string_replace_map:2: WARNING: Enumerated list ends without a blank line; unexpected unindent.

Reply:
Oh yes, I was missing a line-continuation char. My Sphinx isn't reporting this error though. Which version have you got?

Reply:
Actually, I've realised I don't need a line-continuation character (https://stackoverflow.com/questions/54677795/python-and-sphinx-bullet-point-list-in-multiline-google-style-docstring) so I don't understand what's wrong. Adding one doesn't alter the rendered text for me, but then I don't get an error message either! I've changed to using Sphinx's `#.` markup for numbered list items just to be sure but that doesn't change anything either. Maybe it will for you...

Reply:
`sphinx-build -version` gives me `3.2.1` which is rather old. Anyway the extra line sorts out the error.